### PR TITLE
[vNext Tokens] Convert PillButtonBar to the new token system

### PR DESCRIPTION
**Pull request by huwilkes**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes
Convert the PillButtonBar to the new token system. Also removed a handful of the tokens (corner radius, hover colors) since they didn't seem to be used anywhere.

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| PillButtonBar_Before | PillButtonBar_After |

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/911)

### CODE REVIEW

File: ios/FluentUI/Pill Button Bar/PillButtonBar.swift [hunk 1]
```diff
@@ -220,25 +160,82 @@ open class PillButtonBar: UIScrollView, FluentUIWindowProvider {
         buttons[index].isEnabled = true
     }
 
-    open override func layoutSubviews() {
-        super.layoutSubviews()
+    public typealias TokenType = PillButtonBarTokens
```
Comment (huwilkes) [inline ios/FluentUI/Pill Button Bar/PillButtonBar.swift:163]: I know I was just told in the PillButton PR I didn't need this, but Xcode strongly disagreed this time. Not sure what the difference is? I'm able to build PillButtons without the typealias, but PillButtonBar won't have it.

**Comment by huwilkes** [comment]
Still not enough git practice. I rebased it to pick up the changes to TokenizedControl, but now I have a bunch of duplicated commits.

File: ios/FluentUI/Pill Button Bar/PillButtonBar.swift [hunk 2]
```diff
@@ -514,3 +524,11 @@ extension PillButtonBar: UIPointerInteractionDelegate {
         return UIPointerStyle(effect: pointerEffect, shape: nil)
     }
 }
+
+// MARK: PillButtonBarDelegate
+
+@objc(MSFPillButtonBarDelegate)
```
Comment (mischreiber) [inline ios/FluentUI/Pill Button Bar/PillButtonBar.swift:530]: Nit: why is this moved? I generally see delegate APIs specified before their control, though maybe that's my crusty old AppKit-addled brain thinking.

**Review by mischreiber — APPROVED** [review]

File: ios/FluentUI/Pill Button Bar/PillButtonBar.swift (hunk 2 above)
Comment (huwilkes) [reply ios/FluentUI/Pill Button Bar/PillButtonBar.swift:530]: The thinking was that it's internal, not public, so it should be lower. Should I move it back up?

Comment (huwilkes) [reply ios/FluentUI/Pill Button Bar/PillButtonBar.swift:530]: Ah, off by one, public, not open.
